src/app/services/cross-table.service.ts

```diff
@@ -41,7 +42,8 @@ export class CrossTableService {
   private _crossTable: CrossTable;
   private _selectedConstraintCell: GbDraggableCellComponent;
 
-  constructor(private resourceService: ResourceService,
+  constructor(private constraintService: ConstraintService,
```

Comment:
Maybe also add a trigger to update the cohort selection when the cohort selection changes (just as Fractalis has)?
```typescript
this.cohortService.cohortsUpdated.asObservable()
      .subscribe((cohorts: Cohort[]) => {
        this.updateCells()
      });
```